sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -131,8 +145,92 @@ export class ApplicationController extends BaseController {
         "Error while associating the assessment workflow.",
       );
     }
+  }
+
+  /**
+   * Creates a new application draft for the authenticated student.
+   * The student is allowed to have only one draft application, so
+   * this method will create the draft or throw an exception in case
+   * of the draft already exists.
+   * @param payload payload to create the draft application.
+   * @param userToken token from the authenticated student.
+   * @returns the application id of the created draft or an
+   * HTTP exception if it is not possible to create it.
+   */
+  @AllowAuthorizedParty(AuthorizedParties.student)
+  @Post("draft")
+  async createDraftApplication(
+    @Body() payload: SaveApplicationDto,
+    @UserToken() userToken: IUserToken,
+  ): Promise<number> {
+    const programYear = await this.programYearService.getActiveProgramYear(
+      payload.programYearId,
+    );
+
+    if (!programYear) {
+      throw new UnprocessableEntityException(
```

Comment:
Util can be used here as mentioned in previous comment

Reply:
Yes, it could. Please take a look at the previous reply if it makes sense.